addons/mail/models/mail_channel.py

```diff
@@ -266,6 +266,7 @@ def _alias_get_creation_values(self):
     def _subscribe_users(self):
         for mail_channel in self:
             mail_channel.write({'channel_partner_ids': [(4, pid) for pid in mail_channel.mapped('group_ids').mapped('users').mapped('partner_id').ids]})
+        self.channel_last_seen_partner_ids.write({'is_pinned': True})
```

Comment:
I don't like that this is writing `is_pinned` even for pre-existing partners. If/when `is_pinned` will actually be managed in the UI also for channels this would reset it for everyone which is not what we want.

Method should specifically find only the partners that have to be added in the current batch and only apply it to them.

Also, I'm not sure if you did check already, but please report the performance of this method on a database with a lot of users (1k, 10k) and few channels having `group_ids` set, and compare before and after this PR to ensure we didn't make it bad.

Reply:
Also please add the comment I suggested to explain this is necessary to do because writing on `many2many` does not properly apply defaults values from the pivot table model

Reply:
> I don't like that this is writing `is_pinned` even for pre-existing partners. If/when `is_pinned` will actually be managed in the UI also for channels this would reset it for everyone which is not what we want.
> 
> Method should specifically find only the partners that have to be added in the current batch and only apply it to them.

Not totally sure of how you find "the partners that have to be added" ?

> Also, I'm not sure if you did check already, but please report the performance of this method on a database with a lot of users (1k, 10k) and few channels having `group_ids` set, and compare before and after this PR to ensure we didn't make it bad.

How do you do that ?

Reply:
For getting partners that have to be added, naively you just need to subtract those that already members:
`partners_to_add = mail_channel.group_ids.users.partner_id - mail_channel.channel_partner_ids`

And for creating many users you just need to call `create` on `res.users` by giving an array of user data as param. From rpc in JS, or with odoo shell, or with odoo-client-lib, or by creating a python test for it, ... up to you.

And for checking the perf, well it's simple, every time you create a user (with a matching `group_ids`) the method will be called (see `create` override of `res.users` in mail). So once you have a sufficient number of users, you'll see if creating more users is still done in a linear time / very quickly, or if it takes longer and longer due to this method being called.

Reply:
Note that I just realized writing this the method is actually not supporting automatically removing members if their group changes. I wonder if it's done elsewhere or if we just don't care about this.

Reply:
I'll have a look ..

Reply:
I ran following script and seems linear to me.

```python
import timeit

for i in range(10000):
    start = timeit.default_timer()
    env['res.users'].create({"groups_id":[(4,1)],"name":"lol", "login":"lol{}".format(2000+i)})
    stop = timeit.default_timer()
    print('Time: ', stop - start)
```

https://drive.google.com/file/d/1wm94ZBrtBAZizDVCoNynTD7F1zkzkzhn/view

Reply:
> Note that I just realized writing this the method is actually not supporting automatically removing members if their group changes. I wonder if it's done elsewhere or if we just don't care about this.

Looking at the feature and the field,it seems the feature is an auto-add and nothing more, so I found this "normal" that they are not auto removed

Reply:
Ok for the auto-add, I was just curious but we should leave it like that.

Thanks for the perf test, I'll play with it a bit